src/esp/assets/GenericMeshData.cpp

```diff
@@ -36,26 +36,26 @@ void GltfMeshData::uploadBuffersToGPU(bool forceReload) {
   buffersOnGPU_ = true;
 }
 
-Magnum::GL::Mesh* GltfMeshData::getMagnumGLMesh() {
+Magnum::GL::Mesh* GenericMeshData::getMagnumGLMesh() {
   if (renderingBuffer_ == nullptr) {
     return nullptr;
   }
 
   return &(renderingBuffer_->mesh);
 }
 
-void GltfMeshData::setMeshData(Magnum::Trade::AbstractImporter& importer,
-                               int meshID) {
-  ASSERT(0 <= meshID && meshID < importer.meshCount());
+void GenericMeshData::setMeshData(Magnum::Trade::AbstractImporter& importer,
+                                  int meshID) {
+  /* Guarantee mesh instance success */
+  CORRADE_INTERNAL_ASSERT_OUTPUT(meshData_ = importer.mesh(meshID));
   /* Interleave the mesh, if not already. This makes the GPU happier (better
      cache locality for vertex fetching) and is a no-op if the source data is
      already interleaved, so doesn't hurt to have it there always. */
-  Cr::Containers::Optional<Mn::Trade::MeshData> meshData =
-      importer.mesh(meshID);
-  if (meshData)
-    meshData_ = Mn::MeshTools::interleave(*std::move(meshData));
-  else
-    meshData_ = Cr::Containers::NullOpt;
+
+  CORRADE_ASSERT(
+      meshData_->primitive() == Mn::MeshPrimitive::Triangles,
+      "Cannot instance collisionMeshData_; Triangle Mesh expected.", );
+  meshData_ = Mn::MeshTools::interleave(*std::move(meshData_));
```

Comment:
This question got kinda lost in our conversation and still needs to be clarified -- how are the wireframe primitives going to work with this assert in place?

Random thought: since the primitives are, well, primitive, maybe Bullet could get some builtin box/sphere/capsule collision shape in this case instead of a generic "bag of triangles"? That would solve the problem of wireframe primitives also. @aclegg3 is that feasible (also regarding other physic/collision APIs we use)?

Reply:
Yes, we will consume the implicit primitives and built-in shapes when possible. Also, it is not necessary to have triangle meshes to create convex collision shapes (just need a point cloud), so we could relax that requirement a bit for all but the concave scene meshes if necessary.

Reply:
Oh, just a point cloud? In that case the index buffer (or the primitive type) would be unnecessary.

> concave

All primitives currently are convex, but *if* a torus will get ever added, that property won't hold anymore.

Reply:
The default collision handling is going to be implicit for primitives.  Perhaps we can just force all collision handling to be implicit (or at least all wireframe handling) until this can be addressed here?

Reply:
I am fine with forcing built-in collision primitives for the time being. 👍

Reply:
Wow, weird.  I missed a conversation here, sorry about that :)